crates/miden-objects/src/account/component/mod.rs

```diff
@@ -11,6 +13,51 @@ pub use template::*;
 use crate::account::{AccountType, StorageSlot};
 use crate::{AccountError, Word};
 
+// IMPLEMENTATIONS
+// ================================================================================================
+
+impl TryFrom<Package> for AccountComponentTemplate {
+    type Error = AccountError;
+
+    fn try_from(package: Package) -> Result<Self, Self::Error> {
+        let library = package.unwrap_library().as_ref().clone();
+
+        // Extract metadata if present
+        let metadata = match package.account_component_metadata_bytes.as_deref() {
+            Some(metadata_bytes) => AccountComponentMetadata::read_from_bytes(metadata_bytes)
+                .map_err(|err| {
+                    AccountError::other_with_source(
+                        "failed to deserialize account component metadata",
+                        err,
+                    )
+                })?,
+            None => {
+                // Create default metadata for packages without explicit metadata
```

Comment:
Question: should we actually do this? If a package does not have account component metadata, it could be a package for something else entirely (e.g., a note or a tx script).